src/universal_transfer_operator/integrations/hooks/fivetran.py

```diff
@@ -0,0 +1,361 @@
+from typing import Any
+
+import pendulum
+import requests
+from airflow.exceptions import AirflowException
+from airflow.hooks.base import BaseHook
+from requests import exceptions as requests_exceptions
+
+
+class FivetranHook(BaseHook):
+    """
+    Fivetran API interaction hook.
+
+    :param fivetran_conn_id: `Conn ID` of the Connection to be used to configure this hook.
+    :param timeout_seconds: The amount of time in seconds the requests library will wait before timing out.
+    :param retry_limit: The number of times to retry the connection in case of service outages.
+    :param retry_delay: The number of seconds to wait between retries.
+    """
+
+    conn_name_attr = "fivetran_conn_id"
+    default_conn_name = "fivetran_default"
+    conn_type = "fivetran"
+    hook_name = "Fivetran"
+    api_user_agent = "airflow_provider_fivetran/1.1.4"
+    api_protocol = "https"
+    api_host = "api.fivetran.com"
+    api_path_connectors = "v1/connectors/"
+    api_metadata_path_connectors = "v1/metadata/connectors/"
+    api_path_destinations = "v1/destinations/"
+    api_path_groups = "v1/groups/"
+
+    @staticmethod
+    def get_ui_field_behaviour() -> dict:
+        """Returns custom field behaviour"""
+        return {
+            "hidden_fields": ["schema", "port", "extra", "host"],
+            "relabeling": {
+                "login": "Fivetran API Key",
+                "password": "Fivetran API Secret",
+            },
+            "placeholders": {
+                "login": "api key",
+                "password": "api secret",
+            },
+        }
+
+    @staticmethod
+    def _get_airflow_version() -> Any:
+        """
+        Fetch and return the current Airflow version from aws provider
+        https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/hooks/base_aws.py#L486
+        """
+        try:
+            # This can be a circular import under specific configurations.
+            # Importing locally to either avoid or catch it if it does happen.
+            from airflow import __version__ as airflow_version
+
+            return "-airflow_version/" + airflow_version
+        except Exception:
+            # Under no condition should an error here ever cause an issue for the user.
+            return ""
+
+    def __init__(
+        self,
+        fivetran_conn_id: str = "fivetran",
+        fivetran_conn=None,
+        timeout_seconds: int = 180,
+        retry_limit: int = 3,
+        retry_delay: float = 1.0,
+    ) -> None:
+        super().__init__(None)  # Passing None fixes a runtime problem in Airflow 1
+        self.conn_id = fivetran_conn_id
+        self.fivetran_conn = fivetran_conn
+        self.timeout_seconds = timeout_seconds
+        if retry_limit < 1:
+            raise ValueError("Retry limit must be greater than equal to 1")
+        self.retry_limit = retry_limit
+        self.retry_delay = retry_delay
+
+    def _do_api_call(self, endpoint_info, json=None):
```

Comment:
Can we add typing and return types to all the methods added?

Reply:
Please add a code comment.

Reply:
Also, we may not need the whole code, right? we can just inherit the hook class and override the methods/value.

Reply:
This is not required at all once the release is done. I don't think making changes in this code is useful

Reply:
Created issue here: https://github.com/astronomer/apache-airflow-provider-transfers/issues/64